specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/stable/2022-09-04/redhatopenshift.json

```diff
@@ -0,0 +1,1127 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Red Hat OpenShift Client",
+    "description": "Rest API for Azure Red Hat OpenShift 4",
+    "version": "2022-09-04"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/providers/Microsoft.RedHatOpenShift/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "summary": "Lists all of the available RP operations.",
+        "description": "The operation returns the RP operations.",
+        "operationId": "Operations_List",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OperationList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.  If the resource doesn't exist, 404 (Not Found) is returned.  If any of the input parameters is wrong, 400 (Bad Request) is returned.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Lists all of the available RP operations.": {
+            "$ref": "./examples/Operations_List.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.RedHatOpenShift/locations/{location}/listinstallversions": {
+      "get": {
```

Comment:
>get [](http://example.com/codeflow?start=7&length=3)

was this meant to be a POST ? THis current path will be interpreted by arm as a collection get for the resource. If you keep it as a GET the InstallVersions object would need to be a ARM resource.

Reply:
we want it to be a GET request - the idea is for the ARO service to provide same command as AKS, they have `az aks get-versions` and we are working on implementing the `az aro get-versions` equivalent.

I am trying to see more of how AKS achieved this, it looks like the endpoint for their command is as follows:
```
'https://management.azure.com/subscriptions/fe16a035-e540-4ab7-80d9-373fa9a3d6ae/providers/Microsoft.ContainerService/locations/eastus/orchestrators?api-version=2019-04-01&resource-type=managedClusters'
```
is the orchestrators an ARM resource? I see this for AKS service https://github.com/Azure/azure-rest-api-specs/blob/3ec782ff73d2c35f23978d2c6738cc9c85016531/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-06-01/managedClusters.json#L2699-L2703

Reply:
@raosuhas can you review my response here?

Reply:
we are trying to discourage RPs from using this model 

For your scenario you can achieve pretty much the same thing by modelling this as a GET on a collection of "installVersions" i.e 

GET  ...crosoft.RedHatOpenShift/locations/{location}/installversions" 

THis returns a paginated response like so : [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination)

and each entry of the value array is a Proxy resource of the "InstallVersion" type. The only difference from the AKS model is that each installVersion is a Proxy ARM  type.

Reply:
@raosuhas I have refactored my code and api /swagger definition such that this is a `GET` request which returns a `List` of `OpenShiftVersion`'s  that are available.

Can you please review? I do not seem to be able to edit the labels on the PR

Reply:
@ross-bryan I think @raosuhas is also suggesting changing the path to `/installversions`, so in this case the resource type is "InstallVersion"

Reply:
@raosuhas @leni-msft We have changed this resource path to 

GET ...crosoft.RedHatOpenShift/locations/{location}/openshiftversions"